src/onegov/org/views/people.py

```diff
@@ -46,36 +63,46 @@ def view_person(self, request, layout=None):
     }
 
 
-def person_functions_by_organization(subject_person, pages, request):
+def person_functions_by_organization(subject_person, topics, request):
     """ Collects 1:1 mappings of all context-specific functions and
      organizations for a person. Organizations are pages where `subject_person`
      is listed as a person.
 
-     Returns a List of strings in the form:
+     Returns a List of Markup in the form:
 
-        - Organization 1, Function A
-        - Organization 2, Function B
+        - Organization 1: Function A
+        - Organization 2: Function B
+        - ...
 
     This is not necessarily the same as person.function!
     """
 
-    organization_to_function = []
-
-    for topic in pages:
-        people = topic.people
-        for person in people or []:
-            if person.id == subject_person.id:
-                try:
-                    if person.display_function_in_person_directory:
-                        func = person.context_specific_function
-                        if func:
-                            page = f"<a href=\"{request.link(topic)}\">" \
-                                   f"{topic.title}</a>"
-                            organization_to_function.append(
-                                Markup(f"<span>{page}: {func}</span>"))
-                except AttributeError:
-                    continue
-    return organization_to_function
+    TopicFunctionPair = namedtuple("TopicFunctionPair", ["function", "topic"])
```

Comment:
Might as well use `typing.NamedTuple`. That way we don't have to change it once we add type annotations.

Reply:
```suggestion
    class TopicFunctionPair(NamedTuple):
        function: str
        topic: Topic
```

Reply:
You will also need to change the import